packages/neos-ui-editors/src/Library/LinkInput.js

```diff
@@ -150,41 +150,43 @@ export default class LinkInput extends PureComponent {
-        } else if (isEmail(searchTerm)) {
-            this.setState({
-                isLoading: false,
-                searchOptions: [{
-                    label: this.props.i18nRegistry.translate('Neos.Neos.Ui:Main:ckeditor__toolbar__link__formatAsEmail', 'Format as email?'),
-                    loaderUri: `mailto:${searchTerm}`
-                }]
-            });
-        } else if (isUriOrInternalLink(searchTerm)) {
+        if (isUriOrInternalLink(searchTerm)) {
             this.setState({
                 isLoading: false,
                 searchOptions: []
             });
         } else if (searchTerm) {
-            this.setState({isLoading: true, searchOptions: []});
             // We store the searchTerm at the moment lookup was triggered, and only update the options if the search term hasn't changed
             const searchTermWhenLookupWasTriggered = searchTerm;
+            const groupedSearchOptions = [];
+
+            if (looksLikeExternalLink(searchTermWhenLookupWasTriggered)) {
+                groupedSearchOptions.push({
+                    group: this.props.i18nRegistry.translate('Neos.Neos.Ui:Main:ckeditor__toolbar__link__formatOptions', 'Format options'),
+                    label: this.props.i18nRegistry.translate('Neos.Neos.Ui:Main:ckeditor__toolbar__link__formatAsHttp', 'Format as https link?'),
+                    loaderUri: `https://${searchTermWhenLookupWasTriggered}`
+                });
+            } else if (isEmail(searchTermWhenLookupWasTriggered)) {
+                groupedSearchOptions.push({
+                    group: this.props.i18nRegistry.translate('Neos.Neos.Ui:Main:ckeditor__toolbar__link__formatOptions', 'Format options'),
+                    label: this.props.i18nRegistry.translate('Neos.Neos.Ui:Main:ckeditor__toolbar__link__formatAsEmail', 'Format as email?'),
+                    loaderUri: `mailto:${searchTermWhenLookupWasTriggered}`
+                });
+            }
+
+            this.setState({isLoading: true, searchOptions: groupedSearchOptions});
+
             this.props.linkLookupDataLoader.search(this.getDataLoaderOptions(), searchTerm)
                 .then(searchOptions => {
                     if (searchTermWhenLookupWasTriggered === this.state.searchTerm) {
-                        const groupedSearchOption = searchOptions.map(searchOption => {
+                        searchOptions.forEach(searchOption => {
                             searchOption.group = 'assetSourceLabel' in searchOption ? searchOption.assetSourceLabel : this.props.i18nRegistry.translate('Neos.Neos:Main:document');
-                            return searchOption;
+                            groupedSearchOptions.push(searchOption);
                         });
+
                         this.setState({
                             isLoading: false,
-                            searchOptions: groupedSearchOption
+                            searchOptions: [...groupedSearchOptions]
```

Comment:
```suggestion
                            searchOptions: groupedSearchOptions
```

nitpick: is there a reason for this copy?

Reply:
Yep. At line 177 you'll see that `groupedSearchOptions` is already put into state. State then holds a reference to `groupedSearchOptions` which gets mutated in line 184. So to make sure that the changes made to `groupedSearchOptions` after the search results have been fetched also lead to a changed reference, `groupedSearchOptions` needs to be copied at this point.

Reply:
ahh i see, because there would be no diff in setState to the previous one... so technically it would properly cleaner to make a copy already before? But fine with me. Thanks for explaining - i imagined something like this, but its not always super obvious ^^ (i oversaw that 177 happens always and not conditionally ^^)